composite/token-allocation/src/adapter.ts

```diff
@@ -35,19 +36,34 @@ export const marketCapTotalValue = (
     .toNumber()
 }
 
-const inputParams = {
-  allocations: true,
-  quote: false,
-}
-
 const toValidAllocations = (allocations: TokenAllocations): TokenAllocations => {
   if (!allocations.every((t) => !!t.symbol))
     throw new AdapterError({ message: `Symbol not available for all tokens.`, statusCode: 400 })
-  return allocations.map((t) => ({
-    symbol: t.symbol.toUpperCase(),
-    decimals: t.decimals || DEFAULT_TOKEN_DECIMALS,
-    balance: t.balance || DEFAULT_TOKEN_BALANCE * 10 ** (t.decimals || DEFAULT_TOKEN_DECIMALS),
-  }))
+
+  const _toValidDecimals = (decimals: number) =>
+    Number.isInteger(decimals) && decimals >= 0 ? decimals : DEFAULT_TOKEN_DECIMALS
+  const _toValidBalance = (balance: BigNumberish, decimals: number) => {
+    if (!balance) {
+      return DEFAULT_TOKEN_BALANCE * 10 ** decimals
+    }
```

Comment:
I also think we support too much here. We support `number | string | undefined`, which is ok, but then we support both integers and decimal numbers which complicates the API, and validation.

Instead of supporting data like this (coming from synth-index):

```json
{
  "symbol": "BTC",
  "balance": "4.44",
  "decimal": 0
}
```

... let's just throw (or floor the value to 4), and force the client to convert to units.

Because if we continue to support this, then there is no reason not to support something like this:

```json
{
  "symbol": "BTC",
  "balance": "4.44",
  "decimal": 1
}
```

... which would represent the number `0.444`!?